pkgs/stdenv/generic/setup.sh

```diff
@@ -13,35 +13,45 @@ set -o pipefail
 # code). The hooks for <hookName> are the shell function or variable
 # <hookName>, and the values of the shell array ‘<hookName>Hooks’.
 runHook() {
+    local oldOpts="$(shopt -po nounset)"
+    set +u # For hook itself and fallback logic
+
     local hookName="$1"
     shift
     local var="$hookName"
     if [[ "$hookName" =~ Hook$ ]]; then var+=s; else var+=Hooks; fi
+    local -a "var=(\${$var[@]})"
```

Comment:
I did the trailing `[@]` trick before, but `set -u` doesn't like it. Bash is such a mess.

`%Hook`is cool though; I'll definitely use that!